packages/authgear-core/src/container.ts

```diff
@@ -703,25 +732,24 @@ export class _BaseContainer<T extends _BaseAPIClient> {
     }
     let idToken = await this._delegate.tokenStorage.getIDToken(this.name);
     if (!idToken) {
-      throw new AuthgearError("id_token not found");
+      throw new NotAllowedError(
+        "id_token not found. isAppInitiatedSSOToWebEnabled must be true when the user was authenticated."
+      );
     }
     const deviceSecret = await this._delegate.tokenStorage.getDeviceSecret(
       this.name
     );
     if (!deviceSecret) {
-      throw new AuthgearError("device_secret not found");
+      throw new NotAllowedError(
```

Comment:
@louischan-oursky It is named `NotAllowedError` because his current session is not allowed to perform app-initiated-sso-to-web. In your case, the session is also not allowed to perform app-initiated-sso-to-web as it doesn't have the required scope.
I prefer to keep all of these cases as `NotAllowedError` because the app should catch these error and do the same handling, i.e. ask the user to re-login.